### DISCUSSION TITLE
`expo fetch:android:upload-cert` Vs. `eas certificate`

### DISCUSSION THREAD

**Discussion by flexbox** [expo/expo#19034] [Archived: Questions] [answered]
I am in the process of migrating from `expo` to `eas`.

For some reason, the migration broke my Android App Signing.

1. I followed the documentation https://docs.expo.dev/app-signing/app-credentials/#app-signing-by-google-play
2. Contacted the Google Play team
![image](https://user-images.githubusercontent.com/360936/189160801-568dc03f-e509-4a19-b6bf-beec588d0238.png)

They are asking me to provide the `upload_certificate.pem`

I already have done that 2 years ago —following [this thread from the expo forum](https://forums.expo.dev/t/app-signing-certificates-issue-when-we-upload-app-into-play-store/38291/5)

But I have this error

```
❯ expo fetch:android:upload-cert

  $ expo fetch:android:upload-cert is not supported in the local CLI, please use eas credentials instead
```

**Question:**
How can I use `eas credentials` to download the `upload_certificate.pem` file asked by the Google Play Developer Support

**Comment by keith-kurak** [comment, marked as answer]
If you run `eas credentials`, you should be able to choose "Android" and then eventually get to the options to "Manage keystore" and eventually "download an existing key". However, it would also be perfectly fine to generate a new keystore at this point, if you're going to send a new one to Google anyway, so you can use the same CLI to generate a new keystore.

If you'd like to grab the existing keystore from Classic Build, another option is to still use the global CLI by referring to `expo-cli`, e.g., `expo-cli fetch:android:upload-cert`

**Reply by flexbox** [reply]
Hey, thanks @keith-kurak 

After running

```
✔ What do you want to do? › Set up a new keystore
✔ Select build credentials › Build Credentials m6-YvZ3dxQ (Default)
? Generate a new Android Keystore? › (Y/n)
```

I downloaded locally my `.jks` file and I was able to generate my `upload_certificate.pem` with 
```
keytool -export -rfc -alias upload -file upload_certificate.pem -keystore @organisation__app-name.jks
```

**Reply by AmirKhan732** [reply]
eas credentials does not work
